extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.tracer;
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER;
+import static io.opentelemetry.api.trace.SpanKind.SERVER;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+public final class Tracing {
+
+  private static final TextMapGetter<Map<String, String>> TEXT_MAP_GETTER =
+      new TextMapGetter<Map<String, String>>() {
+        @Override
+        public Set<String> keys(Map<String, String> carrier) {
+          return carrier.keySet();
+        }
+
+        @Override
+        @Nullable
+        public String get(@Nullable Map<String, String> carrier, String key) {
+          //noinspection ConstantConditions
+          return carrier == null ? null : carrier.get(key);
+        }
+      };
+
+  private Tracing() {}
+
+  public static void run(String spanName, Runnable runnable) {
+    run(serviceTracer().spanBuilder(spanName).startSpan(), runnable);
+  }
+
+  public static void run(Span span, Runnable runnable) {
+    call(
+        span,
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  /**
+   * Runs a block of code with a new span - ending the span at the end and recording any exception.
+   *
+   * @param spanName name of the new span
+   */
+  public static <T> T call(String spanName, Callable<T> callable) {
+    return call(serviceTracer().spanBuilder(spanName).startSpan(), callable);
+  }
+
+  public static <T> T call(Span span, Callable<T> callable) {
+    return call(span, callable, Tracing::setSpanError);
+  }
+
+  @SuppressWarnings("NullAway")
+  public static <T> T call(
+      Span span, Callable<T> callable, BiConsumer<Span, Exception> handleException) {
+    //noinspection unused
+    try (Scope scope = span.makeCurrent()) {
+      return callable.call();
+    } catch (Exception e) {
+      handleException.accept(span, e);
+      sneakyThrow(e);
+      return null;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Return the tracer to be used in a service
+   *
+   * @return the tracer to be used in a service
+   */
+  public static Tracer serviceTracer() {
+    return GlobalOpenTelemetry.getTracer("service");
```

Comment:
good idea - I've extracted `GlobalTracing` to make that clear

Reply:
> Is there a way we can rename this class to make it super clear that it's utilizing the global, and should only be used in conjunction with the agent?

alternatively, what do you think of injecting an instance of `OpenTelemetry` into `Tracing` (and changing all methods from static to instance)?

Reply:
that is pretty close to what I had with `ExtendedTracer` :+1:

Reply:
I have both now - injetcted `OpenTelemetry` into `Tracing` and a `GlobalTracing` method (which I still find useful).

Reply:
> I have both now - injetcted `OpenTelemetry` into `Tracing` and a `GlobalTracing` method (which I still find useful).

I agree the global can be convenient, but it also creates problems and so we have been trying to eliminate its use from all of our instrumentation

Reply:
the `GlobalTracing` is meant to stay in contrib basically forever, giving users the choice to use the global tracer - in the same way as they will always be able to use `GlobalOpenTelemetry`.

Reply:
I think that currently the recommended use of `GlobalOpenTelemetry` is to use it only once, and inject the resulting `OpenTelemetry` instance wherever you need it. This would probably make a good Java SIG discussion.